client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java

```diff
@@ -98,16 +100,34 @@ public void channelCreated(Channel ch) throws Exception {
             }, channelPoolSize);
         }
 
-        private Single<HttpResponse> sendRequestInternalAsync(final HttpRequest request) {
+        private Single<HttpResponse> sendRequestInternalAsync(final HttpRequest request, Proxy proxy) {
             final URI uri;
             try {
-                uri = new URI(request.url());
+                if (proxy == null) {
+                    uri = new URI(request.url());
+                } else if (proxy.address() instanceof InetSocketAddress) {
+                    InetSocketAddress address = (InetSocketAddress) proxy.address();
+                    String scheme = address.getPort() == 443
+                            ? "https"
+                            : "http";
+
+                    String urlString = scheme + "://"
+                            + ((InetSocketAddress) proxy.address()).getHostString()
+                            + ":" + address.getPort();
+                    uri = new URI(urlString);
```

Comment:
This currently drops any path and/or query parameters that exist in the request's url. I'd recommend using our UrlBuilder to create the url. That should make it easier to construct the proxy url.

Reply:
This uri is just used to acquire a channel, so it only needs the information needed to establish a TCP connection. Do you still want me to append the path and query? I actually considered refactoring the pool to make it use SocketAddress instead of URI, but I didn't want to change any host resolution behaviors @jianghaolu may have intended.

Reply:
For now I'm renaming the variable to "channelAddress" for clarify-- we may wish to change the pool to acquire based on SocketAddress in a separate PR.